hudi-spark/src/main/java/org/apache/hudi/keygen/CustomKeyGenerator.java

```diff
@@ -58,6 +59,7 @@
 
   public CustomKeyGenerator(TypedProperties props) {
     super(props);
+    this.props = props;
```

Comment:
Hey, Since there is a protected `config` field in `KeyGenerator`, which can be shared with all subclasses. Why do not we check that must be non-nullable?

Reply:
+1  there is no need to duplicate this again in the sub class right.

Reply:
@yanghua due to serialization and `config` field being transient, the issue is happening. In CustomKeyGenerator, we try to initialise new object of SimpleKeyGenerator and TimestampBasedKeyGenerator using config object, that is where NPE occurs. The only good way I can think of is to introduce a TimestampBasedKeyGenerator object in CustomKeyGenerator and call `getPartitionPath` method using this object. Let me do the changes here.

Reply:
@pratyakshsharma : Why does config field need to be transient ?

Reply:
@bvaradar It has been like this from the very beginning. I am not sure of the reason though. Making config field non-transient is the easiest fix possible here.